R/conditions.R

```diff
@@ -424,6 +427,7 @@ maybe_lossy_cast <- function(result, x, to,
                              lossy = NULL,
                              locations = NULL,
                              ...,
+                             loss_type = c("precision", "generality"),
```

Comment:
Maybe just make it a string so it's a bit more flexible in the future? I'm not sure "loss of generality" will mean much to most people, and there doesn't seem to be a strong reason to limit the interface, IMO.

Reply:
I think the strong reason is that generation of messages based on arbitrary nouns is very difficult, and even more so when translations are involved. I think we should prefer the stricter interface by default.